src/winerror.cr

```diff
@@ -0,0 +1,2182 @@
+class WinError < Errno
+  # NOTE: `get_last_error` must be called BEFORE an instance of this class
+  # is malloced as it would change the "last error" to SUCCESS
+  def self.new(message)
+    new(message, LibC._GetLastError)
+  end
+
+  def initialize(message, code)
+    buffer = uninitialized UInt8[256]
+    size = LibC._FormatMessageA(LibC::FORMAT_MESSAGE_FROM_SYSTEM, nil, code, 0, buffer, buffer.size, nil)
+    details = String.new(buffer.to_unsafe, size).strip
+    super "#{message}: [WinError #{code}, #{details}]", winerror_to_errno(code)
```

Comment:
Just a tiny nitpick, can we have brackets here (`super("#{message}: [WinError #{code}, #{details}]", winerror_to_errno(code))`), it looks weird without (I actually missed the `winerror_to_errno` call).